controllers/bounce_processes.go

```diff
@@ -334,6 +334,9 @@ func getAddressesForUpgrade(logger logr.Logger, r *FoundationDBClusterReconciler
 	notReadyProcesses := make([]string, 0)
 	addresses := make([]fdbv1beta2.ProcessAddress, 0, len(status.Cluster.Processes))
 	for _, process := range status.Cluster.Processes {
+		if cluster.Spec.DataCenter != "" && cluster.Spec.DataCenter != process.Locality[fdbv1beta2.FDBLocalityDCIDKey] {
```

Comment:
We have to remove this line to make sure the operator is waiting for all processes across all dcs to be ready to be upgraded.